### PR TITLE
refactor: update `blas/ext/base/sfill` to follow current projects conventions

### LINKED ISSUE #1527: [RFC]: refactor `blas/ext/base/sfill` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/sfill`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/sfill) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AgPriyanshu18**
Resolves #1527.

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors `blas/ext/base/sfill` to follow current projects convention

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1527 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@AgPriyanshu18 Would you mind checking the box in the OP that you've read the contributing guidelines? We require this as an acknowledgment of licensing terms and attribution.

File: lib/node_modules/@stdlib/blas/ext/base/sfill/README.md [hunk 1]
```diff
@@ -50,34 +50,30 @@ The function has the following parameters:
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
 -   **stride**: index increment.
 
-The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to fill every other element
+The `N` and `stride` parameters determine which elements in the strided array are accessed at runtime. For example, to fill every other element
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/README.md]:
```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to fill every other element
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/README.md [hunk 2]
```diff
@@ -98,7 +94,7 @@ The function has the following additional parameters:
 
 -   **offset**: starting index.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of `x`
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of the strided array
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/README.md]:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the offset parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of the strided array
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.native.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.ndarray.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.ndarray.native.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/c/Makefile [hunk 7]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/c/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/c/benchmark.length.c [hunk 8]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/benchmark/c/benchmark.length.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/binding.gyp [hunk 9]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/binding.gyp]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/docs/repl.txt [hunk 10]
```diff
@@ -39,19 +39,18 @@
...
     > var x0 = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, -4.0, 5.0, -6.0 ] );
     > var x1 = new {{alias:@stdlib/array/float32}}( x0.buffer, x0.BYTES_PER_ELEMENT*1 );
-    > N = {{alias:@stdlib/math/base/special/floor}}( x0.length / 2 );
-    > {{alias}}( N, 5.0, x1, 2 )
+    > {{alias}}( 3, 5.0, x1, 2 )
     <Float32Array>[ 5.0, 3.0, 5.0, 5.0, 5.0 ]
     > x0
     <Float32Array>[ 1.0, 5.0, 3.0, 5.0, 5.0, 5.0 ]
 
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/docs/repl.txt:53]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/docs/types/index.d.ts [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/docs/types/index.d.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/docs/types/test.ts [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /*
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/docs/types/test.ts]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/ndarray.native.js [hunk 13]
```diff
@@ -46,10 +47,8 @@ var addon = require( './sfill.native.js' );
 */
 function sfill( N, alpha, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+	offset = minViewBufferIndex( N, stride, offset );
+	view = offsetView( x, offset );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/ndarray.native.js:51]:
```suggestion
	offset = minViewBufferIndex( N, stride, offset );

	view = offsetView( x, offset );

```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Minor changes requested. Please avoid updating copyright year. Thanks @AgPriyanshu18 !

File: lib/node_modules/@stdlib/blas/ext/base/sfill/examples/c/Makefile [hunk 14]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/examples/c/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/examples/c/example.c [hunk 15]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/examples/c/example.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/examples/index.js [hunk 16]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/include.gypi [hunk 17]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/include.gypi]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/README.md [hunk 18]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2020 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/README.md]:
```suggestion
Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/include/stdlib/blas/ext/base/sfill.h [hunk 19]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/include/stdlib/blas/ext/base/sfill.h]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/index.js [hunk 20]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/main.js [hunk 21]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/main.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/native.js [hunk 22]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/ndarray.js [hunk 23]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/ndarray.native.js [hunk 24]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/sfill.js [hunk 25]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/sfill.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/lib/sfill.native.js [hunk 26]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/lib/sfill.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/src/Makefile [hunk 27]
```diff
@@ -1,7 +1,7 @@
 #/
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/src/Makefile]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/src/addon.c [hunk 28]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/src/addon.c:4]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/src/addon.c
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/src/addon.c:1]: This file needs to be updated and refactored. It is not just a simple renaming.

File: lib/node_modules/@stdlib/blas/ext/base/sfill/src/sfill.c [hunk 29]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/src/sfill.c]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.js [hunk 30]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.ndarray.js [hunk 31]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.ndarray.native.js [hunk 32]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.sfill.js [hunk 33]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.sfill.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.sfill.native.js [hunk 34]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sfill/test/test.sfill.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

**Review by kgryte — CHANGES_REQUESTED** [review]
I've gone through and undone all the copyright year changes. This PR still needs further updates. Namely, the `addon.c` file needs to be updated and the `manifest.json` file needs to be updated.

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**